### PR TITLE
feat!: refactor module

### PR DESCRIPTION
**Pull request by mfroembgen**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,227 +1,45 @@
...
-  label_value_case    = var.label_value_case
-  descriptor_formats  = var.descriptor_formats
-  labels_as_tags      = var.labels_as_tags
-}
-
-module "redis_label" {
-  count   = var.redis_enabled ? 1 : 0
+module "kvstore_label" {
   source  = "justtrackio/label/null"
   version = "0.26.0"
```
Comment (applike-ss) [inline main.tf:8]: please make sure we can enable/disable the redis/ddb independently

**Review by applike-ss — CHANGES_REQUESTED** [review]

File: examples/all-labels/main.tf [hunk 2]
```diff
@@ -1,19 +0,0 @@
-module "example" {
```
Comment (applike-ss) [inline examples/all-labels/main.tf:1]: why did you remove all examples?

File: variables.tf [hunk 3]
```diff
@@ -1,21 +1,7 @@
-variable "ddb_autoscale_read_schedule" {
```
Comment (applike-ss) [inline variables.tf:1]: why did you remove the ddb scaling schedules?

File: main.tf [hunk 4]
```diff
@@ -1,227 +1,54 @@
...
-  version = "0.66.4"
-
-  context = module.redis_label[0].context
-
-  container_definition_json          = "[${sensitive(module.container_definition[0].json_map_encoded)}]"
+  count  = var.redis_enabled ? 1 : 0
+  source = "github.com/justtrackio/terraform-aws-ecs-redis?ref=add_service_discovery_name_varibale"
+
+  context      = module.kvstore_label.context
+  label_orders = var.label_orders
```
Comment (applike-ss) [inline main.tf:46]: assuming we have a label order of `["environment","stage","name"]`
wouldn't that result in this dynamodb table name `env-stg-name`, while the redis' ecs service is going to be also called like that?
In my opinion, this doesn't make sense, when we have a cluster name of `env`.
Shouldn't we revert to the label order per service again here?

File: main.tf [hunk 5]
```diff
@@ -1,227 +1,54 @@
...
-
-  container_image              = "${var.redis_image_repository}:${var.redis_image_tag}"
-  container_name               = local.redis_container_name
-  container_cpu                = var.redis_cpu_size
-  container_memory_reservation = var.redis_service_memory_size
-
-  port_mappings = [
+module "ddb" {
+  count   = var.ddb_enabled ? 1 : 0
+  source  = "terraform-aws-modules/dynamodb-table/aws"
```
Comment (applike-ss) [inline main.tf:19]: why are you switching away from our own module to this one?

**Review by applike-ss — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 3 above)
Comment (mfroembgen) [reply variables.tf:1]: currently its not supported

File: main.tf (hunk 4 above)
Comment (mfroembgen) [reply main.tf:46]: It uses the label_orders ecs not kvstore. I put it inside this module as well so that we can override it if we want.

File: main.tf (hunk 5 above)
Comment (mfroembgen) [reply main.tf:19]: In this case for consistency as I don't want to use multiple different ddb modules.

File: variables.tf (hunk 3 above)
Comment (mfroembgen) [reply variables.tf:1]: I do need to add a pr for that

File: examples/all-labels/main.tf (hunk 2 above)
Comment (mfroembgen) [reply examples/all-labels/main.tf:1]: Because it will still change a bit and I don't want to waste now time until I know it will stay like that for quite some time

**Review by ztzxt — APPROVED** [review]
